sync/reconcile_pod.go

```diff
@@ -55,19 +61,24 @@ func waitForCacheToContainPod(watchClient *watch.Client, namespace, podName stri
 	return nil, false
 }
 
-func waitForPodContainersToBeReady(watchClient *watch.Client, namespace, podName string, progressResponder func(*client.DeploymentSyncResponse, string)) *v1.Pod {
+func waitForPodContainersToBeReady(watchClient *watch.Client, namespace, podName string, progressResponder func(*client.DeploymentSyncResponse, string)) (*v1.Pod, error) {
 	var statusMessage string
 	for i := 0; i < 15; i++ {
 		if existingPod, ok := watchClient.GetPod(namespace, podName); ok {
 			primary := primaryContainerNameFromPod(existingPod)
 			for _, container := range existingPod.Status.ContainerStatuses {
 				if container.Name == primary && container.Ready {
-					return &existingPod
+					return &existingPod, nil
 				}
 			}
 
-			currentStatusMessage := getPodStatusMessage(existingPod)
+			currentStatusMessage, err := getPodStatusMessage(existingPod)
+			if err != nil {
+				return nil, err
+			}
+
 			if currentStatusMessage != statusMessage {
+				// TODO: does deploymentSyncRequest need to be set for transitioning message?
```

Comment:
No, a transitioning message doesn't need to have a body

Reply:
I'll remove this in a separate PR, it's been this way for a while so it's probably harmless.